thunder/tests/test_inplace_copy.py

```diff
@@ -178,3 +178,11 @@ def func(T0):
     assert_close(a_ref, a)
     for o, o_ref in zip(o_thunder, o_eager):
         assert_close(o, o_ref)
+
+
+@instantiate(dtypes=datatypes.float_math_dtypes)
+def test_inplace_copy_of_leaf_requiring_grad_fails(executor, device, dtype):
+    tdtype = ttorch.to_torch_dtype(dtype)
+    a = make_tensor((4, 4), device=device, dtype=tdtype, requires_grad=True)
+    with pytest.raises(RuntimeError):
+        a.copy_(a)
```

Comment:
I think this is testing that PyTorch raises error when calling inplace on leaf tensor.

It should probably be something like

```python

def fn(a):
    a.copy_(a)

jfn = thunder.jit(fn)

with pytest.raises(RuntimeError, match="a leaf Variable"):
    jfn(a)

```

Reply:
True, whoops.